apps/dashboard/src/components/workflow-editor/configure-workflow.tsx

```diff
@@ -17,16 +17,33 @@ import { useWorkflowEditorContext } from '@/components/workflow-editor/hooks';
 import { cn } from '@/utils/ui';
 import { SidebarContent, SidebarHeader } from '@/components/side-navigation/Sidebar';
 import { PageMeta } from '../page-meta';
+import { ConfirmationModal } from '../confirmation-modal';
 
 export function ConfigureWorkflow() {
+  const [isPauseModalOpen, setIsPauseModalOpen] = useState(false);
   const tagsQuery = useTagsQuery();
   const { isReadOnly } = useWorkflowEditorContext();
 
-  const { control, watch } = useFormContext<z.infer<typeof workflowSchema>>();
+  const { control, watch, setValue } = useFormContext<z.infer<typeof workflowSchema>>();
   const workflowName = watch('name');
 
+  const onPauseWorkflow = () => {
+    setValue('active', false, { shouldValidate: true, shouldDirty: true });
+  };
+
   return (
     <>
+      <ConfirmationModal
+        open={isPauseModalOpen}
+        onOpenChange={setIsPauseModalOpen}
+        onConfirm={() => {
+          onPauseWorkflow();
+          setIsPauseModalOpen(false);
+        }}
+        title="Proceeding will pause the workflow"
+        description="This workflow cannot be triggered if paused, please confirm to proceed."
+        confirmButtonText="Proceed"
+      />
```

Comment:
Can we extract and reuse this confirmation modal component? There's duplicate code